modules/data.land/R/load_veg.R

```diff
@@ -0,0 +1,107 @@
+##' Function uses load_data{benchmark} to get veg data
+##' @name load_veg
+##' @title load_veg
+##' @export
+##' @author Istem Fer
+load_veg <- function(new_site, start_date, end_date, 
+                     source_id, source, icmeta = NULL, format_name = NULL, 
+                     machine_host, dbparms, outfolder, overwrite = FALSE, ...){
+  
+  
+  bety <- dplyr::src_postgres(dbname   = dbparms$bety$dbname, 
+                              host     = dbparms$bety$host, 
+                              user     = dbparms$bety$user, 
+                              password = dbparms$bety$password)
+
+  #--------------------------------------------------------------------------------------------------#
+  # Load data : this step requires DB connections 
+  
+  # query data.path from source id [input id in BETY]
+  query      <- paste0("SELECT * FROM dbfiles where container_id = ", source_id)
+  input_file <- db.query(query, con = bety$con)
+  data_path  <- file.path(input_file[["file_path"]], input_file[["file_name"]])
+  
+  # query format info
+  format <- query.format.vars(bety = bety, input.id = source_id)
+  
+  # load_data{benchmark}
+  obs <- load_data(data.path = data_path, format, site = new_site)
+  
+  #--------------------------------------------------------------------------------------------------#
+  # Match species : this step requires DB connections 
+  
+  if("species_USDA_symbol" %in% format$vars$bety_name){
+    code.col    <- "species_USDA_symbol"
+    format_name <- "usda"
+  }else if("latin_name" %in% format$vars$bety_name){
+    # not encountered an actual case yet, put here as a reminder
+    code.col <- "latin_name"
+    # might indicate a custom format, should be passed to function
+    if(is.null(format_name)){
+      logger.severe("Can't match code to species. Please provide 'match.format' via settings.")
+    }
+  }else{
+    logger.severe("Can't match code to species. No valid format found.")
+  } 
+  
+  # match code to species ID
+  # no lower case
+  obs[[code.col]] <- toupper(obs[[code.col]])
+  
+  # filter dead trees
+  obs <- remove_dead_trees(obs, code.col)
+  spp.info <- match_species_id(input_codes = obs[[code.col]], format_name = format_name, bety = bety)
+  
+  # merge with data
+  tmp <- spp.info[ , colnames(spp.info) != "input_code"]
+  
+  # a hack for now to have a similar structure as the FIA case
+  veg_info      <- list() 
+  
+  if(!is.null(icmeta)){
+    # the first sublist can be for the metadata maybe?
+    # to be handled by veg2model later
```

Comment:
Ok, so remind me, have we generalized the load_data call here to load separate files that have plot info and merge them together as part of the wide2long? Or do we need to handle plot info files separately?  This is a challenging design question since we don't know a priori how many (and which) formats will keep plot info in a separate file vs how many will just have that info as columns in one large file that also includes the tree info.